service/models.py

```diff
@@ -24,4 +27,36 @@ class UprnMapping(db.Model):  # type: ignore
     uprn = db.Column(db.String(20), primary_key=True)
     lr_uprn = db.Column(db.String(20), nullable=False)
 
-Index('uprn_mapping_pkey', UprnMapping.uprn)
+
+class UserSearchAndResults(db.Model):  # type: ignore
+    """
+    Store details of user view (for audit purposes) and update after payment (for reconciliation).
+    """
+
+    # As several users may be searching at the same time, we need a compound primary key.
+    # Note that WebSeal prevents a user from being logged in from multiple places concurrently.
+    search_datetime = db.Column(db.DateTime(timezone=True), nullable=False, primary_key=True)
```

Comment:
as per comment above. needs to include the changes to viewed_time and new column for insertion time.